wirecard/oxid-ee
# WDPD 90: Config Order Delete on Fail/Cancel

### This PR

* Adds config settings to all payment methods to automatically delete an order if the associated transaction failed or is canceled

### Notes

* This is based on WDPD-28

### How to Test

* Go to _Shop Settings → Payment Methods_ and select a Wirecard payment
* Make sure both "Delete Canceled Order" and "Delete Failed Order" is set to "Yes"
* Set "Payment Action" to "Authorize", so that the resulting transaction can be canceled

#### Testing canceled

* Go to the frontend and make a purchase using the payment method
* Go to _Administer Orders → Orders_ and confirm that the order shows up in the table
* Go to _Wirecard Transactions → Transactions_ and select the resulting transaction
* Select the tab "Post-Processing options" and click "Cancel"
* Go back to _Administer Orders → Orders_ and confirm that the order was deleted

#### Testing failed

* Go to the frontend and make a purchase using the payment method
* Make the transaction fail (I don't know how to do this in the interface, but I'm sure there is a way)
* Check the logs and confirm that the associated order was deleted

### Jira Links

* WDPD-90: https://jira.parkside.at/browse/WDPD-90
